lib/internal/test_runner/harness.js

```diff
@@ -233,6 +244,11 @@ function setupProcessState(root, globalOptions) {
 
   const terminationHandler = () => {
     exitHandler();
+
+    if (!root.reported) {
+      process.exitCode = kGenericUserError;
+    }
```

Comment:
nit: MHO, this part is unclear, should we add a comment to explain why it's necessary?